onnxruntime/contrib_ops/cpu/quantization/qlinear_softmax.cc

```diff
@@ -0,0 +1,373 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "contrib_ops/cpu/quantization/qlinear_softmax.h"
+
+#include <cstdint>
+#include <type_traits>
+#include <utility>
+
+#include "core/common/common.h"
+#include "core/providers/common.h"
+#include "core/providers/cpu/tensor/transpose.h"
+
+#include "core/mlas/inc/mlas.h"
+#include "core/platform/threadpool.h"
+#include "gsl/gsl-lite.hpp"
+
+
+namespace onnxruntime {
+namespace contrib {
+
+constexpr int OPSET13 = 13;
+
+namespace {
+
+// concept enabled in cpp20
+template <typename T>
+constexpr bool ValidType = std::is_same_v<T, int8_t> || std::is_same_v<T, uint8_t>;
```

Comment:
What's the purpose of this given we only register int8_t and uint8_t kernels?

i.e. in what scenario do we need BuildLookupTableIfFixed?

Assuming we don't actually need it, we can minimize the binary size with something like the below. Minimizes templatized code. Utilizes a local static for one-off init. The calling code can use a constexpr std::is_same<T, int8_t> and the compiler can throw away the unused path for each calling site. 

```c++
namespace {
std::vector<uint32_t> QlinearBuildLookupTableUint32(const float x_scale,
                                                    size_t reduce_len,
                                                    bool is_signed) {
  std::vector<uint32_t> lookup_table;
  lookup_table.reserve(256);

  const double qscale = fmin(static_cast<double>(UINT32_MAX) / static_cast<double>(reduce_len),
                             static_cast<double>(0x7fffff));
  for (int32_t i = 0; i < 256; ++i) {
    double scaled_exp_xi = qscale * exp(static_cast<double>(i - 255) * static_cast<double>(x_scale));
    // we can't get the real max number of input tensor here, so we just assume 255.
    // in the process of computation, all numbers will have a shift to align 255
    // if is_signed: 1 2 3 ......126 127 -128 -127 ..... -3 -2 -1
    uint8_t index = static_cast<uint8_t>(is_signed ? i - 128 : i);
    lookup_table[index] = static_cast<uint32_t>(lrint(scaled_exp_xi));
  }
}

gsl::span<const uint32_t> GetLookupTable(const float x_scale,
                                         size_t reduce_len,
                                         bool is_signed) {
  if (is_signed) {
    static auto signed_lookup_table = QlinearBuildLookupTableUint32(x_scale, reduce_len, is_signed);
    return signed_lookup_table;
  } else {
    static auto unsigned_lookup_table = QlinearBuildLookupTableUint32(x_scale, reduce_len, is_signed);
    return unsigned_lookup_table;
  }
}
}  // namespace
```

Reply:
If x_scale is fixed, then we only need to build lookup_table once.
But if it's dynamic, then we have to build it every time. 

It's true for us to get rid of `template`, and `is_signed` is enough